### PR TITLE
Fix link format in demo lessons

### LINKED ISSUE #1598: Broken links in demo_lessons.md 🔗💔
### What is the problem?

Some of the links are broken in the [recommended lessons list](https://github.com/carpentries/instructor-training/blob/main/learners/demo_lessons.md). E.g., https://datacarpentry.org/python-ecology-lesson/04-data-types-and-format/
Leads to 404 instead of
https://datacarpentry.org/python-ecology-lesson/04-data-types-and-format.html

### Location of problem (optional)

https://github.com/carpentries/instructor-training/blob/main/learners/demo_lessons.md

### PR DESCRIPTION
**Pull request by ndporter**
Closes #1598.

Replace all episode links containing trailing `/` with full link (`.html`) or remove `/` for compatibility with workbench.

More information:
Previously, links to episodes could end in one of 3 ways and resolve correctly. For example any of the following should have worked:

- `https://swcarpentry.github.io/git-novice-es/02-setup.html`
- `https://swcarpentry.github.io/git-novice-es/02-setup/`
- `https://swcarpentry.github.io/git-novice-es/02-setup`

With the upgrade to workbench, the second (trailing `/`) no longer resolves.

### CODE REVIEW

**Comment by ndporter** [comment]
@carpentries/instructor-training-maintainers Can someone please check that all links resolve correctly before merging? Thanks!

File: learners/demo_lessons.md
Comment (karenword) [inline learners/demo_lessons.md:82]:
```suggestion
| [Data Wrangling and Processing](https://datacarpentry.org/wrangling-genomics/) |  \* [Assessing Read Quality](https://datacarpentry.org/wrangling-genomics/02-quality-control.html) <br/> \* [Trimming and Filtering](https://datacarpentry.org/wrangling-genomics/03-trimming.html) <br/> \* [Automating a Variant Calling Workflow](https://datacarpentry.org/wrangling-genomics/05-automation.html)  | none |  The other episodes depend on actions taken earlier in the lesson - consult [_Teaching Episodes with Dependencies_](#dependencies). <br/>If using [Assessing Read Quality](https://datacarpentry.org/wrangling-genomics/02-quality-control.html), begin with `gunzip SRR2584863_1.fastq.gz`  |
```

File: learners/demo_lessons.md
Comment (karenword) [inline learners/demo_lessons.md:136]:
```suggestion
| [Programming with Python](https://swcarpentry.github.io/python-novice-inflammation/) | any other episode  | [Debugging](https://swcarpentry.github.io/python-novice-inflammation/11-debugging.html) |  |
```

**Comment by karenword** [comment]
(apparently *setup* pages do still resolve with the training / and don't resolve with .html)

File: learners/demo_lessons.md
Comment (dpshelio) [inline learners/demo_lessons.md:138]:
```suggestion
| [R for Reproducible Scientific Analysis](https://swcarpentry.github.io/r-novice-gapminder/) | any other episode  | \* [Writing Good Software](https://swcarpentry.github.io/r-novice-gapminder/16-wrap-up.html)  |  |
```

File: learners/demo_lessons.md
Comment (dpshelio) [inline learners/demo_lessons.md:140]:
```suggestion
| [Using Databases and SQL](https://swcarpentry.github.io/sql-novice-survey/) | any other episode  | [Data Hygiene](https://swcarpentry.github.io/sql-novice-survey/08-hygiene.html) |  |
```

File: learners/demo_lessons.md
Comment (dpshelio) [inline learners/demo_lessons.md:141]:
```suggestion
| [Automation and Make](https://swcarpentry.github.io/make-novice) |  \* [Introduction](https://swcarpentry.github.io/make-novice/01-intro.html)<br/> \*  [Makefiles](https://swcarpentry.github.io/make-novice/02-makefiles.html) | \* [Conclusion](https://swcarpentry.github.io/make-novice/09-conclusion.html) | Later episodes depend on actions taken earlier in the lesson - consult [_Teaching Episodes with Dependencies_](#dependencies). |
```

File: learners/demo_lessons.md
Comment (dpshelio) [inline learners/demo_lessons.md:142]: Ideally... I'd use the trailing `/`, at it's happening behind the scenes. It goes to a `index.html` page. Or to keep consistency with all, to use `/index.html`.

**Review by dpshelio — CHANGES_REQUESTED** [review]
Suggesting some changes to keep consistency. I'll vote for all to include the `.html` which is the pattern you've used the most in your update. For style lessons, I'd recommend either ending on `/` or including the `/index.html`.

**Comment by ndporter** [comment]
@zkamvar Which formats are most robust over time? We should just choose those right now and stick with it (and maybe share with the maintainer community in general).

- For setup/lesson main page: `https://domain/dir/` | `https://domain/dir/index.html'
- For episodes: `https://domain/dir/episode` | `https://domain/dir/episode.html`

**Comment by zkamvar** [comment]
> @zkamvar Which formats are most robust over time? We should just choose those right now and stick with it (and maybe share with the maintainer community in general).
> 
> * For setup/lesson main page: `https://domain/dir/` | `https://domain/dir/index.html'
> * For episodes: `https://domain/dir/episode` | `https://domain/dir/episode.html`

I would go with the more explicit form of `dir/episode.html` and `dir/index.html`

**Comment by ndporter** [comment]
@tobyhodges [Mercurial](https://swcarpentry.github.io/hg-novice/index.html) is still in here but I see it's not workbench, isn't listed on SWC lessons, and is stale from 2018. Can you confirm whether it should be in the demos list?

**Comment by ndporter** [comment]
All links tested and updated, along with some outdated titles. Awaiting response on whether to include Mercurial. Otherwise, should be ready unless someone else wants to check again.

**Review by ndporter — COMMENTED** [review]
Trying to resolve review requests from @dpshelio but having issues. Hopefully this clears the hold.

**Pull request merged**
